cmd/azbrowse/main.go

```diff
@@ -212,20 +238,69 @@ func main() {
 			armclient.PopulateResourceAPILookup(ctx)
 			status.Status("Done getting provider data", false)
 
-			list.SetSubscriptions(subRequest)
+			newList := []*handlers.TreeNode{}
+			for _, sub := range subRequest.Subs {
+				newList = append(newList, &handlers.TreeNode{
+					Display:        sub.DisplayName,
+					Name:           sub.DisplayName,
+					ID:             sub.ID,
+					ExpandURL:      sub.ID + "/resourceGroups?api-version=2018-05-01",
+					ItemType:       handlers.SubscriptionType,
+					SubscriptionID: sub.SubscriptionID,
+				})
+			}
 
+			var newContent string
+			var newTitle string
 			if err != nil {
-				content.SetContent(err.Error(), "Error")
-				return nil
+				newContent = err.Error()
+				newTitle = "Error"
+			} else {
+				newContent = data
+				newTitle = "Subscriptions response"
 			}
-			content.SetContent(data, "Subscriptions response")
+
+			list.Navigate(newList, newContent, newTitle)
+
 			return nil
 		})
 
 		status.Status("Fetching Subscriptions: Completed", false)
 
 	}()
 
+	if navigateToID != "" {
+		go func() {
+			navigatedChannel := eventing.SubscribeToTopic("list.navigated")
+			var lastNavigatedNode *handlers.TreeNode
+
+			for {
+				nodeListInterface := <-navigatedChannel
```

Comment:
Not sure if this is a problem, going to write it down and see. 

If you start with this method but an expander times out then this go routine will carry on running .... listening for `list.navigated` which will be generated if you keep running after the failure and moving around. This could then result in some jumping around later as you landed on the item that starts with the id...

hmm I think this is a real edge case. Maybe a fix would be to subscribe to `statusEvents` and exit the loop if one is generated with a failure but this isn't ideal as a non-essantial handler could have generated the error. 

Maybe simplest is to from from `for` -> `select` with a timeout set so this go routine exists after 45secs if it hasn't got to the item?

Also valid to ignore this.